File: 백준/Gold/1806. 부분합/부분합.py
import sys
input = sys.stdin.readline

if __name__ == "__main__":
    n, s = map(int, input().split())
    arr = list(map(int, input().split()))

    l, r = 0, 0
    cur = 0
    result = 1e9
    while True:
        if cur >= s:
            result = min(result, r - l)
            cur -= arr[l]
            l += 1
        elif r == n:
            break
        else:
            cur += arr[r]
            r += 1
    print(0) if result == 1e9 else print(result)

File: 백준/Gold/1806. 부분합/부분합.java
import java.util.*;
import java.io.*;

public class Main {
	static int n, s;
	static int[] arr;
	public static void main(String[] args) throws Exception{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		n = Integer.parseInt(st.nextToken());
		s = Integer.parseInt(st.nextToken());
		arr = new int[n];

		st = new StringTokenizer(br.readLine());
		for (int i = 0; i < n; i++){
			int tmp = Integer.parseInt(st.nextToken());
			arr[i] = tmp;
		}

		int l = 0, r = 0;
		int sum = arr[0];
		int result = Integer.MAX_VALUE;
		while (l <= r && r < n){
			if (sum >= s){
				result = Math.min(result, r - l + 1);
				sum -= arr[l];
				l += 1;
			}
			else{
				r += 1;
				if (r < n)
					sum += arr[r];
			}
		}
		if (result == Integer.MAX_VALUE){
			System.out.println(0);
		}
		else{
			System.out.println(result);
		}
	}
}